src/core/reactor_backend.cc

```diff
@@ -1612,31 +1612,33 @@ class reactor_backend_uring final : public reactor_backend {
                 return current_exception_as_future<size_t>();
             }
         }
-        class read_completion final : public io_completion {
-            pollable_fd_state& _fd;
-            const size_t _to_read;
-            promise<size_t> _result;
-        public:
-            read_completion(pollable_fd_state& fd, size_t to_read)
-                : _fd(fd), _to_read(to_read) {}
-            void complete(size_t bytes) noexcept final {
-                if (bytes == _to_read) {
-                    _fd.speculate_epoll(EPOLLIN);
+        return readable(fd).then([this, &fd, buffer, len] {
```

Comment:
It's just a nit-pick, but this change is mostly indentation shift, which makes it hard(er) to review. I'd suggest splitting this patch int two -- first: the fix itself, that "breaks" indentation, e.g. by leaving the `return readable(...)` part indented by 2 spaces instead of 4 and leaving the not-touched code where it was; second: "indentation fix" that shifts all lines to their proper indentation positions